nrf_rpc/nrf_rpc.c

```diff
@@ -370,40 +436,111 @@ static void execute_packet(const uint8_t *packet, size_t len)
 	parse_incoming_packet(NULL, packet, len);
 }
 
-static const struct nrf_rpc_group *remote_group_init(struct header *hdr, const uint8_t *packet,
-						     size_t len)
+static bool protocol_version_check(const struct init_packet_data *init_data)
 {
-	void *iter;
-	const struct nrf_rpc_group *group;
-	const char *remote_strid;
-	size_t remote_strid_len;
+	uint8_t protocol_version = NRF_RPC_PROTOCOL_VERSION;
 
-	if (len <= NRF_RPC_HEADER_SIZE) {
-		NRF_RPC_ERR("Init packet does not contain remote group str id");
-		goto error;
+	if ((protocol_version >= init_data->min_version) &&
+	    (protocol_version <= init_data->max_version)) {
+		return true;
 	}
 
-	remote_strid = (char *)&packet[NRF_RPC_HEADER_SIZE];
-	remote_strid_len = len - NRF_RPC_HEADER_SIZE;
+	return false;
+}
+static const struct nrf_rpc_group *remote_group_init(uint8_t src_group_id, const char *strid,
+						     size_t strid_len)
+{
+	void *iter;
+	const struct nrf_rpc_group *group;
 
 	for (NRF_RPC_AUTO_ARR_FOR(iter, group, &nrf_rpc_groups_array,
 				 const struct nrf_rpc_group)) {
-		if ((remote_strid_len == strlen(group->strid)) &&
-		    (memcmp(remote_strid, group->strid, remote_strid_len) == 0)) {
-			group->data->dst_group_id = hdr->src_group_id;
+		if ((strid_len == strlen(group->strid)) &&
+		    (memcmp(strid, group->strid, strid_len) == 0)) {
+			group->data->dst_group_id = src_group_id;
```

Comment:
Do I understand `nrf_rpc_groups_array` contains group names with termination character, and the transmitted `strid` is without termination char?
Is it not possible to transmit a `strid` with termination character? Then it is possible to use the functions for string operations.